test/HealthChecks.AzureServiceBus.Tests/DependencyInjection/AzureEventHubUnitWithTokenRegistrationTests.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        [Fact]
+        public void add_named_health_check_when_properly_configured()
+        {
+            var services = new ServiceCollection();
+            services.AddHealthChecks()
+                .AddAzureEventHub("cnn", "eventHubName", new AzureCliCredential(),
+                name: "azureeventhubcheck");
+
+            using var serviceProvider = services.BuildServiceProvider();
+            var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+            var registration = options.Value.Registrations.First();
+            var check = registration.Factory(serviceProvider);
+
+            registration.Name.ShouldBe("azureeventhubcheck");
+            check.ShouldBeOfType<AzureEventHubHealthCheck>();
+        }
+
+        [Fact]
+        public void add_named_health_check_using_factories_when_properly_configured()
+        {
+            var services = new ServiceCollection();
+            bool endpointFactoryCalled = false, eventHubNameFactoryCalled = false, tokenCredentialFactoryCalled = false;
+            services.AddHealthChecks()
+                .AddAzureEventHub(_ =>
+                    {
+                        endpointFactoryCalled = true;
+                        return "cnn";
+                    },
+                    _ =>
+                    {
+                        eventHubNameFactoryCalled = true;
+                        return "eventHubName";
+                    },
+                    _ =>
+                    {
+                        tokenCredentialFactoryCalled = true;
+                        return new AzureCliCredential();
+                    },
+                    "azureeventhubcheck");
+
+            using var serviceProvider = services.BuildServiceProvider();
+            var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+            var registration = options.Value.Registrations.First();
+            var check = registration.Factory(serviceProvider);
+
+            registration.Name.ShouldBe("azureeventhubcheck");
+            check.ShouldBeOfType<AzureEventHubHealthCheck>();
+            endpointFactoryCalled.ShouldBeTrue();
+            eventHubNameFactoryCalled.ShouldBeTrue();
+            tokenCredentialFactoryCalled.ShouldBeTrue();
+        }
+
+        [Fact]
+        public void fail_when_no_health_check_configuration_provided()
+        {
+            var services = new ServiceCollection();
+            services.AddHealthChecks()
+                .AddAzureEventHub(string.Empty, string.Empty, new AzureCliCredential());
+
+            using var serviceProvider = services.BuildServiceProvider();
+            var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+            var registration = options.Value.Registrations.First();
+
+            Assert.Throws<ArgumentNullException>(() => registration.Factory(serviceProvider));
```

Comment:
`Should.Throw<>` + check argument name

Reply:
Should I fix this in the AzureServiceBus* tests as well or should we leave it for later?

Reply:
Up to you.